central/image/datastore/store/common/v2/merge.go

```diff
@@ -55,6 +61,36 @@ func mergeComponents(parts ImageParts, image *storage.Image) {
 	}
 }
 
+func mergeComponentsV2(parts ImageParts, image *storage.Image) {
+	// If the image has a nil scan, there is nothing to fill in.
+	if image.GetScan() == nil {
+		return
+	}
+
+	// Use the edges to combine into the parent image.
+	for _, cp := range parts.Children {
+		if cp.ComponentV2 == nil {
+			log.Errorf("UNEXPECTED: nil component when retrieving components for image %q", image.GetId())
+			continue
+		}
+		// Generate an embedded component for the edge and non-embedded version.
+		image.Scan.Components = append(image.Scan.Components, generateEmbeddedComponentV2(image.GetScan().GetOperatingSystem(), cp))
+	}
+
+	sort.SliceStable(image.GetScan().GetComponents(), func(i, j int) bool {
```

Comment:
why do we need to sort components?

Reply:
That is a very good question.  I sorted it because it was sorted before.  See line 50.  I don't really understand the value of sorting them by name and version.  Looks like it may be for the user experience.

![image](https://github.com/user-attachments/assets/655bc183-5362-416e-9cea-f862d38b415f)

@charmik-redhat any thoughts?

Reply:
Yeah I think it is just for user experience. I don't see any other use case. Same when CVEs are sorted by name in each component.

Reply:
I made the executive decision to remove the sort.  If we later determine we need it we can add it back in.  However, in all likelihood if we needed said sort we would sort it in the query before providing to the converter.